src/features/draw-rolling-circle/Drawing.js

```diff
@@ -0,0 +1,108 @@
+// @ts-check
+
+import { CanvasBuilder } from '@/components/Canvas'
+import { Color, Physics } from '@/config'
+
+const HEIGHT = 100
+const WIDTH = 500
+const radius = 30
+let pointX, pointY
+
+const initialState = {
+  rollingCircle: {
+    theta: -1 * Math.PI / 2,
+    direction: 1,
+    speed: Physics.VELOCITY,
+    centerX: WIDTH / 2,
+    centerY: HEIGHT / 2
+  },
+  cycloid: {
+    x: [WIDTH / 2 + radius * Math.cos(-1 * Math.PI / 2)],
+    y: [WIDTH / 2 + radius * Math.sin(-1 * Math.PI / 2)],
+    i: 0
+  }
+}
+
+// - React Component
+
+export const Drawing = new CanvasBuilder()
+  .withId('DrawRollingCircle')
+  .withHeightAndWidth(HEIGHT, WIDTH)
+  .withInitialDrawState(initialState)
+  .withDrawFactory(drawFactory)
+  .build()
+
+// - Draw Factory
+
+/** @type {import('@/components/Canvas/CanvasBuilder').DrawFactory} */
+function drawFactory (ctx, { drawState, setDrawState }) {
+  const rollingCircle = { ...drawState.rollingCircle }
+  const cycloid = { ...drawState.cycloid }
+  let requestId
+
+  function draw (drawArgs) {
+    const { isRollingCirclePaused } = drawArgs
+
+    if (rollingCircle.centerX >= WIDTH - radius || rollingCircle.centerX <= radius) {
+      rollingCircle.direction *= -1
+    }
+
+    ctx.clearRect(0, 0, WIDTH, HEIGHT)
+    ctx.save()
+
+    // fill in background, just for reference
+    ctx.fillStyle = Color.BLACK
```

Comment:
You'll need to add `BLACK` to `/config.colors.js`

Reply:
Funny - it works!  I'll do this though.